source/installation/resource-manager/kubernetes.rst

```diff
@@ -265,13 +319,87 @@ for more details.
+
+The policies enforcing UID/GID and supplemental groups are utilizing the `k8-ldap-configmap`_ tool that generates ConfigMap resources based on LDAP data.  This tool runs as a deployment inside the Kubernetes cluster.
+
 Enforcing Walltimes
 *******************
 
-TODO docs about the job-pod-reaper and k8-namespace-reaper
+In order to enforce that OnDemand pods have a finite runtime it's necessary to deploy a service that can cleanup pods that have run past their walltime.  Also because OnDemand is bootstrapping a namespace per user it's useful to cleanup unused namespaces.
+
+The OnDemand pods will have the ``pod.kubernetes.io/lifetime`` annotation set that is read by `job-pod-reaper`_ that will kill pods that have reached their walltime.  The `job-pod-reaper`_ service runs as a Deployment inside Kubernetes and will kill any pods based on the lifetime annotation.  Below is an example of Helm values to use configure job-pod-reaper for OnDemand:
+
+.. code-block:: yaml
+
+  reapNamespaces: false
+  namespaceLabels: app.kubernetes.io/name=open-ondemand
+  objectLabels: app.kubernetes.io/managed-by=open-ondemand
+
+You will need to tell OnDemand you are using `job-pod-reaper`_ and to bootstrap the necessary RoleBinding so that
+service can delete OnDemand pods. Update ``/etc/ood/config/hooks.env`` to include the following configuration:
+
+.. code-block:: sh
+
+  USE_JOB_POD_REAPER="true"
+
+In order to cleanup unused namespaces the `k8-namespace-reaper`_ tool can be used.  This tool will delete a namespace based on several factors:
+
+- The creation timestamp of the namespace
+- ``openondemand.org/last-hook-execution`` annotation set by OnDemand hook
+- The last pod to run in that namespace based on Prometheus metrics
+
+Below is an example of Helm values to deploy this tool for OnDemand where the OnDemand namespaces have ``user-`` prefix:
+
+.. code-block:: yaml
+
+  config:
+    namespaceLabels: app.kubernetes.io/name=open-ondemand
+    namespaceRegexp: user-.+
+    namespaceLastUsedAnnotation: openondemand.org/last-hook-execution
+    prometheusAddress: http://prometheus.prometheus:9090
+    reapAfter: 8h
+    lastUsedThreshold: 4h
+    interval: 2h
+
+Using a private image registry
+******************************
+
+OnDemand's Kubernetes integration can be setup to pull images from a private registry like `Harbor <https://goharbor.io/>`_.
+
+In order to pull images from a private registry that requires authentication, OnDemand can be configured to setup `Image Pull Secrets <https://kubernetes.io/docs/tasks/configure-pod-container/pull-image-private-registry/>`_.
+The OnDemand web node will need a JSON file setup that includes the username and password of a registry user authorized
+to pull images used by OnDemand apps.
+
+In the following example you can set the following values:
```

Comment:
This is readable by the user right? If so it should be noted here.

Reply:
Yep, I added a warning block to recommend a limited access and read-only token be used.